store/badger.go

```diff
@@ -59,6 +71,18 @@ func (b *BadgerKV) Close() error {
 	return b.db.Close()
 }
 
+func (b *BadgerKV) gc(period time.Duration, discardRatio float64, logger types.Logger) {
+	ticker := time.NewTicker(period)
+	defer ticker.Stop()
+	for range ticker.C {
```

Comment:
add graceful shutdown handling
and probably godoc
```
case <-db.ctx.Done():
    // Exit the function if the database context is cancelled
    return
}

Reply:
channel added as an exit case

Reply:
I thought the db has exposed ctx.
from the docs:
```
// Only one GC is allowed at a time. If another value log GC is running, or DB
// has been closed, this would return an ErrRejected.
```

maybe it's more elegenat to just check for this type of error

Reply:
but if we check for ErrRejected, it maybe the previous didnt finish for some reason, for instance, and we kill the loop while it was not necessary...

Reply:
fine
Even though I'm not sure how `maybe the previous didnt finish` is possible. does `RunValueLogGC` actually runs in the background?

Reply:
yes, sorry, this shoudlnt happen, but from the docs not sure if ErrRejected will only happen when closing the db...